src/client/datascience/jupyter/liveshare/hostJupyterServer.ts

```diff
@@ -371,7 +376,16 @@ export class HostJupyterServer extends LiveShareParticipantHost(JupyterServerBas
                 `Compute Launch Info uri = ${resource?.fsPath}, changed ${changedKernel}, ${launchInfo.kernelConnectionMetadata?.id}`
             );
         }
+        if (!changedKernel && kernelConnection && kernelConnection.id !== launchInfo.kernelConnectionMetadata?.id) {
```

Comment:
I think this will fix a lot of the issues.
We search for kernel connection, and if its different we don't change the kernel connection, hence when using Jupyter (non-raw), we end up using the same interpreter as the one used to start Jupyter.

Reply:
Good find. That's been around for a while then.